spk/minio/src/wizard/upgrade_uifile.sh

```diff
@@ -13,57 +13,104 @@ cat <<EOF > $SYNOPKG_TEMP_LOGFILE
-			  {
-				"key": "wizard_data_directory",
-				"desc": "Data directory location",
-				"defaultValue": "${WIZARD_DATA_DIRECTORY}",
-				"validator": {
-				  "allowBlank": false,
-				  "regex": {
-					"expr": "/^\\\/volume[0-9]{1,2}\\\/[^<>: */?\"]*/",
-					"errorText": "Path should begin with /volume?/ where ? is volume number (1-99)"
-				  }
-				}
-			  },
-			  {
-				"key": "wizard_access_key",
-				"desc": "MinIO access key",
-				"defaultValue": "${WIZARD_ACCESS_KEY}",
-				"validator": {
-				  "allowBlank": false,
-				  "minLength": 3,
-				  "regex": {
-					"expr": "/^[^<>:*/?\"|]*$/",
-					"errorText": "Not allowed character in key"
-				  }
-				}
-			  },
-			  {
-				"key": "wizard_secret_key",
-				"desc": "MinIO secret key",
-				"defaultValue": "${WIZARD_SECRET_KEY}",
-				"validator": {
-				  "allowBlank": false,
-				  "minLength": 8,
-				  "regex": {
-					"expr": "/^[^\"|]*$/",
-					"errorText": "Not allowed character in key"
-				  }
-				}
-			  }
-			]
-		  }
+			{
+                "type": "combobox",
+                "desc": "Please select a volume to use for the data folder",
+                "subitems": [
+                    {
+                        "key": "wizard_data_volume",
+                        "desc": "Volume name",
+						"defaultValue": "${WIZARD_DATA_VOLUME}",
+                        "displayField": "display_name",
+                        "valueField": "volume_path",
+                        "editable": false,
+                        "mode": "remote",
+                        "api_store": {
+                            "api": "SYNO.Core.Storage.Volume",
+                            "method": "list",
+                            "version": 1,
+                            "baseParams": {
+                                "limit": -1,
+                                "offset": 0,
+                                "location": "internal"
+                            },
+                            "root": "volumes",
+                            "idProperty": "volume_path",
+                            "fields": [
+                                "display_name",
+                                "volume_path"
+                            ]
+                        },
+                        "validator": {
+                            "fn": "{console.log(arguments);return true;}"
+                        }
+                    }
+                ]
+            },
+			{
+                "type": "textfield",
+                "desc": "Data shared folder (using the volume chosen above)",
+                "subitems": [
+                    {
+                        "key": "wizard_data_directory",
+                        "desc": "Data shared folder",
+                        "defaultValue": "${WIZARD_DATA_DIRECTORY}",
+                        "validator": {
+                            "allowBlank": false,
+                            "regex": {
+                                "expr": "/^[\\w _-]+$/",
```

Comment:
you must escape the `\` to fix the upgrade wizard: to get `\\` you must write `\\\\`
```suggestion
                                "expr": "/^[\\\\w _-]+$/",
```